diff --git a/CMakeLists.txt b/CMakeLists.txt
index d8c45bbe..aa7201b3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,6 +59,17 @@ if(CMAKE_BUILD_TYPE MATCHES Debug OR CMAKE_BUILD_TYPE MATCHES DEBUG)
 else()
   add_compile_options(-O3)
   message(STATUS "Configuring Hyprland in Release with CMake")
+
+  include(CheckIPOSupported)
+  check_ipo_supported(RESULT IPO_SUPPORTED OUTPUT IPO_ERROR LANGUAGES C CXX)
+
+  if(IPO_SUPPORTED)
+      message(STATUS "IPO / LTO enabled")
+  else()
+      message(WARNING "IPO / LTO not supported: <${IPO_ERROR}>")
+  endif()
+
+  set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ${IPO_SUPPORTED})
 endif()
 
 include_directories(. "src/" "subprojects/udis86/" "protocols/")
diff --git a/meson.build b/meson.build
index 6a9b7ac5..8d98a986 100644
--- a/meson.build
+++ b/meson.build
@@ -5,6 +5,7 @@ project('Hyprland', 'cpp', 'c',
     'default_library=static',
     'optimization=3',
     'buildtype=release',
+    'b_lto=true',
     'debug=false',
     'cpp_std=c++26',
   ])
